src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePicker.cs

```diff
@@ -196,62 +193,79 @@ public DatePicker()
 
 			DefaultStyleKey = typeof(DatePicker);
 
+			this.Loaded += DatePicker_Loaded;
+			this.Unloaded += DatePicker_Unloaded;
+
 			InitPartial();
 
 			PrepareState();
 		}
 
-		~DatePicker()
-		{
-			// This will ensure the pending async operation
-			// completes, closed the open dialog, and doesn't
-			// try to execute a callback to a DatePicker that
-			// no longer exists.
-			if (m_tpAsyncSelectionInfo != null)
-			{
-				/*VERIFYHR*/
-				m_tpAsyncSelectionInfo.Cancel();
-			}
+#if HAS_UNO // TODO Uno specific: Window activation handling to accomodate for WinUI multiwindow, original implementation can be ported when DatePicker is updated from WinUI 3 sources.
+		private readonly SerialDisposable _windowActivatedToken = new();
 
-			m_epWindowActivatedHandler.Disposable = null;
+		private void DatePicker_Unloaded(object sender, RoutedEventArgs e)
+		{
+			_windowActivatedToken.Disposable = null;
 		}
 
-		// Initialize the DatePicker
-		void PrepareState()
+		private void DatePicker_Loaded(object sender, RoutedEventArgs e)
 		{
-			Window pCurrentWindow = global::Microsoft.UI.Xaml.Window.Current;
-
-			// DatePickerGenerated.PrepareState();
-
-			// We should update our state during initialization because we still want our dps to function properly
-			// until we get applied a template, to do this we need our state information.
-			UpdateState();
-
-			if (pCurrentWindow != null)
+			// TODO: Uno Specific: This portion of code was originally in PrepareState,
+			// but was moved here as it requires XamlRoot for multiwindow purposes.
+			if (XamlRoot.HostWindow is { } window)
 			{
 				WeakReference wrWeakThis = new WeakReference(this);
 
-				pCurrentWindow.Activated += (s, pArgs) =>
-				{
+				window.Activated += OnWindowActivated;
+				_windowActivatedToken.Disposable = Disposable.Create(() => window.Activated -= OnWindowActivated);
 
+				void OnWindowActivated(object sender, WindowActivatedEventArgs args)
+				{
 					DatePicker spThis;
 
 					spThis = wrWeakThis.Target as DatePicker;
 					if (spThis != null)
 					{
 						CoreWindowActivationState state =
 							CoreWindowActivationState.CodeActivated;
-						state = (pArgs.WindowActivationState);
+						state = (args.WindowActivationState);
 
 						if (state == CoreWindowActivationState.CodeActivated
 							|| state == CoreWindowActivationState.PointerActivated)
 						{
 							spThis.RefreshSetup();
 						}
 					}
-				};
+				}
 			}
 		}
+#endif
+
+		~DatePicker()
```

Comment:
This is likely to fail if not scheduled on the dispatcher.

Reply:
As this is already in master, I will handle this separately in https://github.com/unoplatform/uno/issues/15224